src/cider/util/CiderBitUtils.h

```diff
@@ -170,6 +170,44 @@ FORCE_INLINE void clearBitAt(uint8_t* bit_vector, size_t index) {
   bit_vector[index >> 3] &= kCiderBitReverseMask[index & 0x7];
 }
 
+FORCE_INLINE bool isBitClearAt(const uint8_t* bit_vector, size_t index) {
+  // return bit_vector[index >> 3] ^ kCiderBitMask[index & 0x7];
+  return bit_vector[index >> 3] ^ (1 << (index & 0x7));
+}
+
+FORCE_INLINE void setBitAt1(uint8_t* bit_vector, size_t index, bool is_null) {
+  // https://stackoverflow.com/questions/24314281/conditionally-set-or-clear-bits
+  // https://graphics.stanford.edu/~seander/bithacks.html#ConditionalSetOrClearBitsWithoutBranching
+  uint8_t idx = index >> 3;
+  uint8_t mask = 1 << (index & 0x7);
+  bit_vector[idx] = (bit_vector[idx] & ~mask) | (-(uint8_t)is_null & mask);
+}
+/*
+FORCE_INLINE uint8_t setBitAt1(uint8_t* bit_vector,
```

Comment:
remove this? and do we have any performance data on this optimization?

Reply:
performance degrades instead， this code removes the branch successfully but generates some extra i8 to i32 and i32 to i8 opcodes.